contracts/tg4-stake/src/contract.rs

```diff
@@ -158,7 +168,31 @@ pub fn execute_bond<Q: CustomQuery>(
         .add_attribute("action", "bond")
         .add_attribute("amount", amount)
         .add_attribute("sender", &info.sender);
-    res.messages = update_membership(deps.storage, info.sender, new_stake, &cfg, env.block.height)?;
+
+    // Update the sender's vesting stake
+    let new_vesting_stake =
+        STAKE_VESTING.update(deps.storage, &info.sender, |stake| -> StdResult<_> {
+            Ok(stake.unwrap_or_default() + vesting_amount)
+        })?;
+    // Delegate (stake to contract) to sender's vesting account
+    if vesting_amount > Uint128::zero() {
+        let msg = TgradeMsg::Delegate {
+            funds: coin(vesting_amount.into(), cfg.denom.clone()),
+            staker: info.sender.to_string(),
+        };
+        res = res
+            .add_message(msg)
+            .add_attribute("vesting_amount", vesting_amount);
+    }
+
+    // Update membership messages
+    res = res.add_submessages(update_membership(
+        deps.storage,
+        info.sender,
+        new_stake + new_vesting_stake,
```

Comment:
Tried to optimise this, but the fact is that I need to load the data for the current stake (even when not modified), because of the update membership call.

If I don't load the values, the membership points get messed up on stake update. Hopefully there was a test already for that.